### PR TITLE
Gangmageddon

### PR DESCRIPTION
**Pull request by ghost**
@Robustin's lovechild revived

🆑 Robustin
add: New gamemode -- Gangmageddon, a variant of Gang which disables Security, and introduces the Vigilante role in lieu of security.
add: Vigilantes have their own modified gangtool that gets influence for surviving, being implanted, destroying gang equipment, and removing tags.
add: Vigilante gangtools have their own unique lineup of weaponry and gear to help them battle the gang menace.
add: During Gangmageddon, the carnage starts early - so vigilantes will get waves of reinforcements in the form of "posses" that show up every ~12m or so. Gangs will have a reinforcement gateway that can tip the balance of power in your gang's favor, assuming you can keep it operational.
add: Gangs now have access to the revivification serum, a simple way to bring back fallen gangsters, and other non-combat equipment. Sandbags are now available for bosses in the security version of the mode.
balance: Much of the station's weaponry and WMD's are removed during the no-sec version of gang mode.
/🆑

### CODE REVIEW

**Comment by mememanrino** [comment]
you are literally hitler for doing this. i hope you know this.

**Comment by enthethan** [comment]
i expected nothing less

**Comment by Manly-Dwarf** [comment]
horrible gamemode that'll be run like once and never again

File: hippiestation/code/modules/antagonists/gang/vigilante.dm [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+/datum/antagonist/vigilante/on_removal()
+	. = ..()
+	if(owner && owner.current)
+		for(var/obj/item/gangtool/hell_march/vigilante/O in owner.current.contents)
+			qdel(O)
+
+/datum/antagonist/vigilante/greet()
+	if(!owner.current)
+		return
+	to_chat(owner.current, "<FONT size=3><u><b>You are a Vigilante!</b></u><br> Nanotrasen has given all loyal crew the authority to eliminate gang activity aboard the station.<br> You possess a reverse-engineered gangtool that rewards influence for destroying gangster equipment.<br> You will also receive influence for keeping the station free of gang tags.<br><b>Prevent gangs from taking over the station!<b></FONT>")
```
Comment (yoyobatty) [inline hippiestation/code/modules/antagonists/gang/vigilante.dm]: Might be a good idea to enforce the "no murderboning station" line

Comment (ghost) [reply hippiestation/code/modules/antagonists/gang/vigilante.dm]: Done

`Use lethal force against gangsters <i>if they cannot be converted back to Nanotrasen</i>, but do not kill loyal crewmembers!`

**Comment by AnthonyJNgo** [comment]
Test merge is broken right now? @steamp0rt Resolve conflicts please, and bug someone for a review.

**Review by yoyobatty — APPROVED** [review]
Looks ok.

**Pull request merged**